sharedbuffers/mapped_struct.py

```diff
@@ -1309,19 +1301,28 @@ def iteritems(self):
             yield key, self.vlist[idx]
 
     def items(self):
-        return list(self.iteritems())
+        if python3:
+            return self.iteritems()
+        else:
+            return list(self.iteritems())
 
     def iterkeys(self):
         return iter(self.objmapper)
 
     def keys(self):
-        return self.objmapper.keys()
+        if python3:
+            return iter(self.objmapper)
```

Comment:
Why not just 

```suggestion
            return self.objmapper
```

It's iterable, so it's a valid return value (py3 returns an iterable set-like object not a pure iterator)

We're missing the set-iike part but maybe we can fix it in objmapper in some future PR.